### PR TITLE
cli: strip the --background flag

### LINKED ISSUE #6449: extra --background flag
When you use `--background` to start up a node in the background, everything works fine, but when you look at the process, the flag is shown a second time with `background=false`, which is confusing.

```
JESSEs-MacBook-Pro:cockroach jesseseldess$ ./cockroach start --background
build:     beta-20160428-94-g1b81f2c @ 2016/05/02 15:00:43 (go1.6)
admin:     http://localhost:8080
sql:       postgresql://root@localhost:26257?sslmode=disable
logs:      cockroach-data/logs
store[0]:  path=cockroach-data

JESSEs-MacBook-Pro:cockroach jesseseldess$ ps
  PID TTY           TIME CMD
 1752 ttys000    0:00.07 -bash
 3970 ttys000    0:14.60 ruby /usr/local/bin/jekyll serve
 3975 ttys000    0:00.01 /usr/local/lib/ruby/gems/2.2.0/gems/rb-fsevent-0.9.7/bin/fsevent_watch --latency 0.1 /Users/jesseseldess/docs
 1928 ttys001    0:00.14 -bash
 4005 ttys001    0:00.12 ./cockroach start --background --background=false
 3909 ttys002    0:00.04 -bash
```

@bdarnell

### PR DESCRIPTION
**Pull request by petermattis**
Fixes #6449.

---

This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/7028)

### CODE REVIEW

**Comment by mberhault** [comment]
LGTM

File: cli/start.go [hunk 1]
```diff
@@ -264,11 +265,18 @@ func runStart(_ *cobra.Command, _ []string) error {
...
+	foundBackground := false
+	for _, arg := range os.Args {
+		if arg == "--background" || strings.HasPrefix(arg, "--background=") {
+			foundBackground = true
+			continue
+		}
+		args = append(args, arg)
+	}
+	if !foundBackground {
+		args = append(args, "--background=false")
```
Comment (tschottdorf) [inline cli/start.go]: why even append this? Seems more coherent to always strip out the flag.

Comment (mberhault) [inline cli/start.go]: if `--background` wasn't specified on the commandline, it means the default is true somehow, so we need to override it with false.

Comment (bdarnell) [inline cli/start.go]: If we're in `rerunBackground`, then the background flag must have been present in a way that the flag package was able to parse, but our simplistic prefix matching didn't find it. This is a safeguard against the flag parsing rules being more complicated than we understand.

One thing we're missing here is that the string `--background` could be used as the argument to another flag: `cockroach start --store --background --background` is legal (if nonsensical) and creates a data directory named `--background` while running in the background. This code would strip out both occurrences of `--background`.

This sort of parsing bug sets of my security senses, although I'm not able to come up with any problematic scenarios other than someone inflicting pain on themselves by deliberately choosing weird names for things.

**Comment by bdarnell** [comment]
LGTM

**Pull request merged**